src/state_machine.cpp

```diff
@@ -22,96 +18,127 @@ void StateMachine::construct()
   State& sit_start = this->createGaitState(SIT, SIT_SELECTED, SIT_ACTIVATED, "gait_sit", &turn_off_start);
 
   // Sub-menus
-  this->constructObstacleMenu(&obstacles, &walk, &single_step, &obstacles);
-  this->constructWalkMenu(&walk, &obstacles);
-  this->constructStepMenu(&single_step, &obstacles);
+  this->constructObstacleMenu(&obstacles);
+
+  // Determine the last added state before the walk and single step menu are added
+  // This is used in constructing the walk and single step menu
+  std::list<State>::iterator last_state = states_.end();
```

Comment:
[`list::end`](https://www.cplusplus.com/reference/list/list/end/) returns an iterator referring to the object that follows the last object in the list. In your comment you describe it as it returns the last element. [`list::back`](https://www.cplusplus.com/reference/list/list/back/) returns the actual last element in a list if that is what you were looking for.

Reply:
Ah I see, the code is right, but the comment is wrong indeed. I will fix the comment